lib/date_range.rb

```diff
@@ -0,0 +1,56 @@
+require 'date'
+
+module Hotel
+  class DateRange
+    attr_reader :check_in, :check_out
+    
+    def initialize(check_in, check_out)
+      @check_in = check_in
+      @check_out = check_out
+      
+      validate_date_range
+      validate_date_object
+    end
+    
+    
+    def validate_date_range
+      if @check_out < @check_in
+        raise ArgumentError, "End date cannot be before start date"
+      end
+    end
+    
+    
+    def validate_date_object
+      unless @check_in.class == Date && @check_out.class == Date
+        raise ArgumentError, "Please enter a valid date!"
+      end
+    end    
+    
+
+    def num_nights
+      num_nights = (@check_out - @check_in).to_i
+      return num_nights
+    end  
+    
+    
+    # User story: I can access the list of reservations for a specific date, so that I can track reservations by date
+    def date_in_range?(date)
+      range = @check_in...@check_out
+      
+      if range.include?(date)
+        return true # there is an existing reservation for tonight
+      else
+        return false # there are no existing reservations for tonight
+      end
```

Comment:
You can just return the condition directly since it's already a boolean value:

```suggestion
      # true: there is an existing reservation for tonight
      # false: there are no existing reservations for tonight
      return range.include?(date)
```